src/components/PopupWithForm.js

```diff
@@ -20,10 +25,31 @@ class PopupWithForm extends Popup {
     });
     return inputValues;
   }
+  //this where the code for changing the save button to say "Saving..." goes
+
+  saveInfoListener() {
+    this._submitButton = this.popupForm.querySelector(".modal__button");
+    this._submitButton.textContent = "Saving...";
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`